### DISCUSSION TITLE
Typescript reference don't work

### DISCUSSION THREAD

**Discussion by suunryo** [vitest-dev/vitest#5397] [Q&A]
```
/// <reference types="vitest" />
import { defineConfig } from 'vite'

export default defineConfig({
  test: {
    // ...
  },
})
```

Usage of 'reference' is not valid for my project, typescript prompts me that 'test' is not in type 'UserConfigExport'

Package version:
vite: 5.0.12
vitest: 1.3.1
typescript: 5.4.2

**Comment by hi-ogawa** [comment]
You can also try using `defineConfig` from `vitest/config`:

https://github.com/vitest-dev/vitest/blob/6dca36ab877c6ac8653fef7d38386dc1156337b1/test/snapshots/test/fixtures/custom-serializers/vitest.config.ts#L1-L3

**Comment by Habibulloh08** [comment]
/// <reference types="vitest" />
/// <reference types="vite/client" />
import { defineConfig } from "vite";
import react from "@vitejs/plugin-react";

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [react()],
  test: {
    globals: true,
    environment: "jsdom",
    setupFiles: "./src/tests/setup.ts",
  },
});

try writing these codes in vitest config file if the problem is not solved https://vitest.dev/config/
you can get the answer from vite page

**Comment by ka2n** [comment]
Check and ensure that node_modules/vitest/node_modules/vite does not exist.
Duplicated package can cause type mismatches.

**Comment by sopost-kirt** [comment]
Turned out for me it was indeed to do with duplicate packages causing type mismatches as mentioned by ka2n.

This is because I had:
node_modules/vite @ v6
node_modules/vitest/node_modules/vite @ v5

Vitest is using vite at v5, so clearly it's not compatible with v6. You just need to install a vite version that's compatible with the latest version of vitest (at time of writing):

```bash
bun uninstall vite && bun add vite@^5
```

Now your vitest node_modules folder still has the old version of vite installed, so if you simply re-install vitest it will work:

```bash
bun uninstall vitest && bun install vitest -D
```
